File: python/course-repo/solution/01-durable-job-queue/src/webhooks/workflow.py
from datetime import timedelta

from temporalio import workflow

# Pass-through imports: bring activity + shared types into the workflow
# without pulling them into the Python SDK's per-task workflow sandbox reload.
with workflow.unsafe.imports_passed_through():
    from .activities import deliver_webhook
    from .shared import WebhookDelivery


@workflow.defn
class WebhookWorkflow:
    """Tiny Workflow wrapper around deliver_webhook. The standalone caller in
    send_standalone.py and this Workflow both use the same Activity function —
    the "same code, two job types" demo lands fully in Module 06.
    """

    @workflow.run
    async def run(self, req: WebhookDelivery) -> int:
        return await workflow.execute_activity(
            deliver_webhook,
            req,
            start_to_close_timeout=timedelta(seconds=10),
        )

File: typescript/course-repo/exercise/01-durable-job-queue/src/workflow.ts
import { proxyActivities } from '@temporalio/workflow';
import type { WebhookDelivery } from './shared';
import type * as activities from './activities';

const { deliverWebhook } = proxyActivities<typeof activities>({
  startToCloseTimeout: '10 seconds',
});

// Thin Workflow that calls deliverWebhook as a step.
// The same Activity the standalone caller submits directly.
// Not used in Module 01's narrative - Module 06 uses this pattern.
export async function webhookWorkflow(req: WebhookDelivery): Promise<number> {
  return await deliverWebhook(req);
}
